dsa-23au/java-dsa/pswish_insertion_app/src/main/java/com/pswishcorp/app/HeapSort.java

```diff
@@ -0,0 +1,58 @@
+package com.pswishcorp.app;
+
+public class HeapSort {
+    /**
+     * Method for HeapSort 
+     * @ param 1 the index of an item in the heap
+     * @ return the index of the left child 
+     * */
+    
+
+    private static int leftChild (int i) {
+        return 2 * i +1 ;
+    }
+
+    /**
+     * Method for heapsort that is used in deleteMax and buildHeap.
+     * @param a an array of Comparable items.
+     * @int i the position from which to percolate down.
+     * @int n the logical size of the binary heap.  
+     */
+
+    private static <AnyType extends Comparable<? super AnyType>> void percDown (AnyType [] a, int i, int n){
+        AnyType tmp;
+        int child;
+        for (tmp = a[i]; leftChild(i) < n; i = child){
```

Comment:
This code looks like it favors the left child, or percolating down toward the left (like depth-first search). Would it ever be appropriate in a heap to percolate to the right?